common/arch_type.cc

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "common/arch_type.h"
+
+#include <cassert>
+
+#include "common/platform.h"
+
+namespace cdc_ft {
+
+static constexpr char kUnhandledArchType[] = "Unhandled arch type";
+
+ArchType GetLocalArchType() {
+  // TODO(ljusten): Take CPU architecture into account.
+#if PLATFORM_WINDOWS
+  return ArchType::kWindows_x86_64;
+#elif PLATFORM_LINUX
+  return ArchType::kLinux_x86_64;
+#endif
+}
+
+bool IsWindowsArchType(ArchType arch_type) {
+  switch (arch_type) {
+    case ArchType::kWindows_x86_64:
+      return true;
+    case ArchType::kLinux_x86_64:
+      return false;
+    default:
+      assert(!kUnhandledArchType);
+      return false;
+  }
+}
+
+bool IsLinuxArchType(ArchType arch_type) {
+  switch (arch_type) {
+    case ArchType::kWindows_x86_64:
+      return false;
+    case ArchType::kLinux_x86_64:
+      return true;
+    default:
+      assert(!kUnhandledArchType);
+      return false;
+  }
+}
+
+const char* GetArchTypeStr(ArchType arch_type) {
```

Comment:
This could return a `std::string_view`.

Reply:
I get that it's better to take a std::string_view as input parameter in functions that don't modify the string. Is there any advantage in this case? Functions that take a std::string would convert the input in any case. Functions that take a const char* or a string view would be roughly equally efficient, give or take (cheap) string_view construction.

Have you been using string views consistently in Orbit? I have tried to use absl string views in the past, but I always found them to be a hassle to use. IIRC conversion was weird and painful, e.g. there was no implicit const char* conversion or similar, so you couldn't simply call a function that takes a const char*.

Reply:
We moved everything to string_view in Orbit and avoid const char* as much as possible.

You're right. C-APIs that need a const char* are a pain because string_view is not null-terminated so you have to construct a temporary string and then call `c_str` to get a proper C-string. Google's style guide is quite clear though that this is preferred over not using std::string_view.

That being said, feel free to ignore that here. Just a suggestion for the future I guess.

Reply:
I guess that would be a good candidate for a large-scale refactoring.